src/api/c/meanshift.cpp

```diff
@@ -18,39 +18,39 @@
 using af::dim4;
 using namespace detail;
 
-template<typename T, bool is_color>
-static inline af_array mean_shift(const af_array &in, const float &s_sigma, const float &c_sigma, const unsigned iter)
+template<typename T, bool IsColor>
```

Comment:
Could you pass IsColor as an argument until it is required by the kernel? There is no reason to instantiate all these functions for something that is only needed by the kernel.